### PR TITLE
Deprecate additionalHelperClassNames

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: docs/contributing/muzzle.md [hunk 1]
```diff
@@ -30,6 +30,11 @@ the `InstrumentationModule#isHelperClass(String)` predicate). Aside from referen
 the collection process also builds a graph of dependencies between internal instrumentation helper
 classes - this dependency graph is later used to construct a list of helper classes that will be
 injected to the application classloader (`InstrumentationModule#getMuzzleHelperClassNames()`).
+Muzzle also automatically generates the `InstrumentationModule#getMuzzleContextStoreClasses()`
+method.
+
+If you extend any of these `getMuzzle...()` methods in your `InstrumentationModule`, the muzzle
+compile plugin will not override your code: muzzle will skip those custom methods.
```
Comment (iNikem) [inline docs/contributing/muzzle.md:37]: This sounds like muzzle with skip MY custom methods.

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java [hunk 2]
```diff
@@ -110,10 +119,11 @@ public void visit(
     public FieldVisitor visitField(
         int access, String name, String descriptor, String signature, Object value) {
       if (MUZZLE_REF_MATCHER_FIELD_NAME.equals(name)) {
-        // muzzle field has been generated
-        // by previous compilation
-        // ignore and recompute in visitEnd
-        return null;
+        generateReferenceMatcherField = false;
+        log.warn(
```
Comment (iNikem) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java:123]: `warn` sounds like somebody has to do something about it. Is it the case here?

Comment (mateuszrzeszutek) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java:123]: True, `info` will be more appropriate

File: docs/contributing/muzzle.md (hunk 1 above)
Comment (mateuszrzeszutek) [reply docs/contributing/muzzle.md]: Reworded

**Review by trask — APPROVED** [review]
👍

**Review by iNikem — APPROVED** [review]

**Pull request merged**